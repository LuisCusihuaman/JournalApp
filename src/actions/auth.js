import { types } from "../types/types";
import { firebase, googleAuthProvider } from "../firebase/firebase-config";
import { startLoading, finishLoading } from "./ui";
export const startLoginEmailPassword = (email, password) => {
	return (dispatch) => {
		dispatch(startLoading());
		firebase
			.auth()
			.signInWithEmailAndPassword(email, password)
			.then(({ user }) => {
				dispatch(login(user.uid, user.displayName));
				dispatch(finishLoading());
			})
			.catch((e) => {
				console.log(e);
				dispatch(finishLoading());
			});
	};
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
	return (dispatch) => {
		firebase
			.auth()
			.createUserWithEmailAndPassword(email, password)
			.then(async ({ user }) => {
				await user.updateProfile({ displayName: name });
				dispatch(login(user.uid, user.displayName));
			})
			.catch((e) => console.log(e));
	};
};

export const startGoogleLogin = () => {
	return (dispatch) => {
		firebase
			.auth()
			.signInWithPopup(googleAuthProvider)
			.then(({ user }) => {
				dispatch(login(user.uid, user.displayName));
			});
	};
};

export const login = (uid, displayName) => ({
	type: types.login,
	payload: {
		uid,
		displayName,
	},
});
