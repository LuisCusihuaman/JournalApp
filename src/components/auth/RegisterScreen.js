import React from "react";
import { Link } from "react-router-dom";
import validator from "validator";
import { useForm } from "../../hooks/useForm";

export const RegisterScreen = () => {
	const [formValues, handleInputChange] = useForm({
		name: "Luis!",
		email: "test@test.com",
		password: "123456",
		password2: "123456",
	});
	const { name, email, password, password2 } = formValues;

	const isFormValid = () => {
		if (name.trim().length === 0) {
			console.log("Name is required");
			return false;
		} else if (!validator.isEmail(email)) {
			console.log("Email is not valid");
			return false;
		} else if (password !== password2 || password.length < 5) {
			console.log("Password should be at least 6 characters and match");
			return false;
		}

		return true;
	};
	const handleRegister = (e) => {
		e.preventDefault();
		if (isFormValid()) {
			console.log("formulario correcto");
		}
	};
	return (
		<>
			<h3 className="auth__title">Register</h3>
			<form onSubmit={handleRegister}>
				<div className="auth__alert-error">Hola Mundo</div>
				<input
					type="text"
					placeholder="Name"
					name="name"
					className="auth__input"
					autoComplete="off"
					value={name}
					onChange={handleInputChange}
				/>
				<input
					type="text"
					placeholder="Email"
					name="email"
					className="auth__input"
					autoComplete="off"
					value={email}
					onChange={handleInputChange}
				/>
				<input
					type="password"
					placeholder="Password"
					name="password"
					className="auth__input"
					value={password}
					onChange={handleInputChange}
				/>
				<input
					type="password"
					placeholder="Confirm"
					name="password2"
					className="auth__input"
					value={password2}
					onChange={handleInputChange}
				/>
				<button
					type="submit"
					className="btn btn-primary btn-block mb-5"
				>
					Register
				</button>
				<Link to="/auth/login" className="link mt-5">
					Already registered?
				</Link>
			</form>
		</>
	);
};
