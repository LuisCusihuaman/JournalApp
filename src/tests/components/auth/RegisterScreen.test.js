import "@testing-library/jest-dom";
import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { RegisterScreen } from "../../../components/auth/RegisterScreen";
import { types } from "../../../types/types";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
	auth: {},
	ui: {
		loading: false,
		msgError: null,
	},
};

let store = mockStore(initState);
// store.dispatch = jest.fn();

const wrapper = mount(
	<Provider store={store}>
		<MemoryRouter>
			<RegisterScreen />
		</MemoryRouter>
	</Provider>
);

describe("Pruebas en <RegisterScreen />", () => {
	test("debe de mostrarse correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});

	test("deberia lanzar un error la accion cuando el correo es incorrecto", () => {
		const emailField = wrapper.find('input[name="email"]');

		emailField.simulate("change", {
			target: {
				value: "",
				name: "email",
			},
		});

		wrapper.find("form").simulate("submit", {
			preventDefault() {},
		});

		const actions = store.getActions();

		expect(actions[0]).toEqual({
			type: types.uiSetError,
			payload: "Email is not valid",
		});
	});
});
