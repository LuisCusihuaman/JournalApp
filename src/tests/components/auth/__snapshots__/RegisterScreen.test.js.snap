// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RegisterScreen /> debe de mostrarse correctamente 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Luis!"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="test@test.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm"
      type="password"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link mt-5"
      href="/auth/login"
      onClick={[Function]}
    >
      Already registered?
    </a>
  </form>,
]
`;
